src/framework/global/progress.h

```diff
@@ -37,8 +37,11 @@ struct Progress
     async::Channel<int64_t /*current*/, int64_t /*total*/, std::string /*title*/> progressChanged;
     async::Channel<ProgressResult> finished;
 
+    async::Notification cancelRequested;
```

Comment:
May be 'canceled'?
Because the action has already happened

Reply:
Not really... I've added a comment to the code to clarify.